src/axom/quest/MeshViewUtil.hpp

```diff
@@ -0,0 +1,778 @@
+{
+namespace internal
+{
+template <typename T>
+constexpr T BADINDEX = std::numeric_limits<T>::max();
+template <typename T, int DIM>
+inline axom::StackArray<T, DIM> makeStackArray(T v = std::numeric_limits<T>::max())
+{
+  axom::StackArray<T, DIM> rval;
+  for(int d = 0; d < DIM; ++d)
+  {
+    rval[d] = v;
+  }
+  return rval;
+}
+template <typename T, int DIM, typename U>
+inline axom::StackArray<T, DIM> makeStackArray(const U* v)
+{
+  axom::StackArray<T, DIM> rval;
+  for(int d = 0; d < DIM; ++d)
+  {
+    rval[d] = v[d];
+  }
+  return rval;
+}
+}  // namespace internal
+
+/**
+   \brief Utility for high-level access into a blueprint mesh,
+   for structured mesh with explicit coordinates.
+
+   Blueprint mesh data is sufficient but sparse, leaving users
+   to compute a number of intermediate data to get to high-level
+   data of interest, such as views into array data.  This class
+   encapsulates those common utilities.
+
+   Views are single-domain-specific.  They don't apply to multi-domain
+   meshes.  They are also topology specific, with the topology name
+   given to the constructor.
+
+   TODO: Figure out if there's a better place for this utility.
+   It's only in axom/quest because the initial need was there.
+*/
+template <int DIM, axom::MemorySpace MemSpace>
+class MeshViewUtil
+{
+public:
+  using CoordsViewsType =
+    axom::StackArray<axom::ArrayView<double, DIM, MemSpace>, DIM>;
+  using ConstCoordsViewsType =
+    axom::StackArray<axom::ArrayView<const double, DIM, MemSpace>, DIM>;
+
+  //@{
+  //@name Setting up
+  /*!
+    @brief Constructor
+    @param [in] bpDomain Blueprint single domain.
+    @param [in] topologyName Name of topology in the domain.
+
+    If \a topologyName is omitted, use the first topology,
+
+    The topology dimension must match DIM.
+  */
+  MeshViewUtil(conduit::Node& bpDomain, const std::string& topologyName = "")
+    : m_dom(&bpDomain)
+    , m_topologyName(topologyName)
+  {
+    if(m_topologyName.empty())
+    {
+      m_topologyName = bpDomain.fetch_existing("topologies")[0].name();
+    }
+
+    const std::string topoPath = "topologies/" + m_topologyName;
+    m_topology = &m_dom->fetch_existing(topoPath);
+
+    const std::string coordsetPath =
+      "coordsets/" + m_dom->fetch_existing(topoPath + "/coordset").as_string();
+    m_coordset = &m_dom->fetch_existing(coordsetPath);
+
+    m_cdom = m_dom;
+    m_ctopology = m_topology;
+    m_ccoordset = m_coordset;
+
+    if(!isValid())
+    {
+      SLIC_ERROR(
+        "Invalid domain in MeshViewUtil.  Domain must be blueprint-valid,"
+        " structured, with exlicit coordinates and dimension equal to "
+        " the template DIM parameter.");
+    }
+
+    computeCoordsDataLayout();
+  }
+
+  /*!
+    @brief Constructor
+    @param [in] bpDomain const Blueprint single domain.
+    @param [in] topologyName Name op topology in the domain.
+  */
+  MeshViewUtil(const conduit::Node& bpDomain, const std::string& topologyName)
```

Comment:
Do we want to overload it like this?
```cpp
MeshViewUtil(conduit::Node& bpDomain, const std::string& topologyName = "")
MeshViewUtil(const conduit::Node& bpDomain, const std::string& topologyName)
```
w/ two separate implementations that might drift?

It might be better to have one constructor that doesn't take a topology name and another that doesn't:
```cpp
MeshViewUtil(conduit::Node& bpDomain)
MeshViewUtil(const conduit::Node& bpDomain, const std::string& topologyName)
```
where the former would call `bpDomain.fetch_existing("topologies")[0].name();` and then invoke the latter. (Perhaps through a utility function).

Reply:
Overloading would lead to 4 constructors for the combinations of with/without `topologyName` and const/non-const `bpDomain`.  Supporting both const and non-const domains is a must.  I'm not sure I see a compelling reason for not using default values for `topologyName`.